[BUGFIX] Affichage du temps majoré dans la liste des candidats de Pix Certif. (PIX-8008)

mariannebost opened the pull request:
## :unicorn: Problème
Un candidat peut être inscrit à une session de certification dans l’onglet Candidats d’une session soit individuellement, soit parmi une liste de candidats.

Dans les deux cas, si le candidat en bénéficie, l’utilisateur Pix Certif doit indiquer le temps majoré qui lui est accordé pour passer sa certification. Cette information est ensuite affichée dans la liste des candidats inscrits dans la session.

Lorsque le pourcentage de temps majoré dépasse 99%, seul le premier chiffre est affiché (1 pour 120%, 2 pour 200%, 9 pour 999%).

## :robot: Proposition
Changer le helper coté ember "format-percentage" pour que si il recoit un chiffre au dessus de 1 il soit gérer de la même manière.

## :100: Pour tester
1. Se connecter à l'app certif avec le compte `certif-sup@example.net`
2. Aller sur une session
3. Ajouter un candidat en important un ods avec une valeur au dessus de 99 pour la colonne "temps majoré"
4. Voir dans le détail du candidat que le pourcentage est bien affiché
5. Voir dans l'espace surveillant que ce même pourcentage est bien affiché également

mariannebost committed:
:recycle: certif: fix helper format percentage with value above 1

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr6173.review.pix.fr)
  * [App (.org)](https://app-pr6173.review.pix.org)
  * [Orga (.fr)](https://orga-pr6173.review.pix.fr)
  * [Orga (.org)](https://orga-pr6173.review.pix.org)
  * [Certif (.fr)](https://certif-pr6173.review.pix.fr)
  * [Certif (.org)](https://certif-pr6173.review.pix.org)
  * [Admin](https://admin-pr6173.review.pix.fr)
  * [API](https://api-pr6173.review.pix.fr/api/)
  * [Pix1D](https://pix1d-pr6173.review.pix.fr)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr6173/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr6173/environment)

aceol commented:
L'arrondi en javascript pose probleme, il faut l'arrondi
<img width="277" alt="Screenshot 2023-05-10 at 16 55 07" src="https://github.com/1024pix/pix/assets/3769147/9df0bceb-be8f-4290-a27a-b3e691e2e156">
<img width="897" alt="Screenshot 2023-05-10 at 16 24 41" src="https://github.com/1024pix/pix/assets/3769147/f04d317e-6703-4021-bb4c-b11660dceb37">

aceol commented on certif/app/helpers/format-percentage.js:
```diff
@@ -2,7 +2,7 @@ import { helper } from '@ember/component/helper';
 export function formatPercentage([value, ..._]) {
   if (value) {
-    const percentageValue = value < 1 ? value * 100 : value;
+    const percentageValue = value * 100;
```
```suggestion
    const percentageValue = Math.round(value * 100);
```

aceol committed:
:sparkles: certif: Round percentage

Floating value are tricky with javascript, we need to round
1.15 * 100 => 1.14999999

Steph0 commented on certif/app/helpers/format-percentage.js:
```diff
@@ -2,7 +2,7 @@ import { helper } from '@ember/component/helper';
-    const percentageValue = value < 1 ? value * 100 : value;
+    const percentageValue = Math.round(value * 100);
 
     return `${Math.trunc(percentageValue)} %`;
```
Vu que l'on utilise `Math.round` plus haut, il n'y a plus besoin de `Math.trunc`

Steph0 replied:
❓ Dans le cadre de la possibilité de la langue anglaise, faudrait-il passer cela en traduction ?
En anglais le formatage devrait être `13%` tandis que en français il y a un espace `13 %`

mariannebost replied:
Vu avec l'équipe un ticket va être créé pour gérer ce cas

aceol replied:
Espace géré dans un autre ticket 👍
